tracer/src/Datadog.Trace/LifetimeManager.cs

```diff
@@ -55,11 +56,32 @@ public static LifetimeManager Instance
             }
         }
 
+        public TimeSpan TaskTimeout { get; set; } = TimeSpan.FromSeconds(30);
+
         public void AddShutdownTask(Action action)
         {
             _shutdownHooks.Enqueue(action);
         }
 
+        public void AddShutdownTask(Func<Task> func)
```

Comment:
nit - As we keep the other one for Security, should we rename this one to `AddAsyncShutdownTask` to make it more explicit

Reply:
done in [43e5a69](https://github.com/DataDog/dd-trace-dotnet/pull/2583/commits/43e5a694657ee3a2c735083ba0b2e8e3a33736bd)